internal/jobservice/eventstojobs/eventstojobs.go

```diff
@@ -0,0 +1,129 @@
+package eventstojobs
+
+import (
+	"context"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+
+	"github.com/G-Research/armada/internal/jobservice/configuration"
+	"github.com/G-Research/armada/internal/jobservice/repository"
+	"github.com/G-Research/armada/pkg/api"
+	js "github.com/G-Research/armada/pkg/api/jobservice"
+	"github.com/G-Research/armada/pkg/client"
+)
+
+type EventsToJobService struct {
+	queue                string
+	jobsetid             string
+	jobid                string
+	jobServiceConfig     *configuration.JobServiceConfiguration
+	jobServiceRepository repository.JobServiceRepository
+}
+
+func NewEventsToJobService(
+	queue string,
+	jobsetid string,
+	jobid string,
+	jobServiceConfig *configuration.JobServiceConfiguration,
+	jobServiceRepository repository.JobServiceRepository) *EventsToJobService {
+	return &EventsToJobService{
+		queue:                queue,
+		jobsetid:             jobsetid,
+		jobid:                jobid,
```

Comment:
We can discuss.

In general, I found that I can't really subscribe from a client grpc call in the way that you would prefer.  If you return a rpc call, it will cancel the go routine once the call returns.  

So I built a subscribe timer that subscribes on a job-set for x amount of time and it will return if you have a terminal state for your job id.